src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorCodeDocument.PropertyTable.cs

```diff
@@ -14,16 +14,26 @@ namespace Microsoft.AspNetCore.Razor.Language;
 public sealed partial class RazorCodeDocument
 {
     /// <summary>
-    ///  Represents a set of mutable values associated with a <see cref="RazorCodeDocument"/>.
+    ///  Represents an immutable set of values associated with a <see cref="RazorCodeDocument"/>.
     /// </summary>
-    private readonly struct PropertyTable()
+    private readonly struct PropertyTable
     {
-        // To add a mutable value, increase Size by 1 and add a new property below.
+        // To add a value, increase Size by 1 and add a new property below.
         // Use a Property<T> for reference types or a BoxedProperty<T> for value types.
 
         private const int Size = 10;
 
-        private readonly object?[] _values = new object?[Size];
+        private readonly object?[] _values;
+
+        public PropertyTable()
+        {
+            _values = new object?[Size];
+        }
+
+        private PropertyTable(object?[] values)
+        {
+            _values = values;
+        }
 
         public Property<IReadOnlyList<TagHelperDescriptor>> TagHelpers => new(_values, 0);
```

Comment:
Super dumb question:

I can understand why we had the array and the Property/BoxedProperty types before, but I'm not understanding why we still need them. Is this somehow better than the brain-dead
``` CSharp
public TagHelperDescriptor? TagHelpers { get; init; }

public PropertyTable WithTagHelpers(TagHelperDescriptor? tagHelpers)
    => Clone() with { TagHelpers = tagHelpers };
```

Reply:
So the simple answer is that I just made this change as mechanically as possible, so didn't touch the structure. I do think we could probably just drop the property table concept altogether at this point and just have objects on the code document itself.